pylatex/tikz.py

```diff
@@ -34,13 +46,462 @@ def __init__(self, options=None, *, data=None):
+                'TikzCoordinate class')
+
+        self._node_text = text
+
+    def dumps(self):
+        """Return string representation of the node."""
+
+        ret_str = []
+        ret_str.append(Command('node', options=self.options).dumps())
+
+        if self.handle is not None:
+            ret_str.append('({})'.format(self.handle))
+
+        if self._node_position is not None:
+            ret_str.append('at {}'.format(str(self._position)))
+
+        if self._node_text is not None:
+            ret_str.append('{{{text}}};'.format(text=self._node_text))
+        else:
+            ret_str.append('{};')
+
+        return ' '.join(ret_str)
+
+    def get_anchor_point(self, anchor_name):
+        """Return an anchor point of the node, if it exists."""
+
+        if anchor_name in self._possible_anchors:
+            return TikZNodeAnchor(self.handle, anchor_name)
+        else:
+            try:
+                anchor = int(anchor_name.split('_')[1])
+            except:
+                anchor = None
+
+            if anchor is not None:
+                return TikZNodeAnchor(self.handle, str(anchor))
+
+        raise ValueError('Invalid anchor name: "{}"'.format(anchor_name))
+
+    def __getattr__(self, attr_name):
+        try:
+            point = self.get_anchor_point(attr_name)
+            return point
+        except ValueError:
+            pass
+
+        # raise AttributeError(
+        #    'Invalid attribute requested: "{}"'.format(attr_name))
+
+
+class TikZUserPath(LatexObject):
+    """Represents a possible TikZ path."""
+
+    def __init__(self, path_type, options=None):
+        """
+        Args
+        ----
+        path_type: str
+            Type of path used
+        options: Options
+            List of options to add
+        """
+        super(TikZUserPath, self).__init__()
+        self.path_type = path_type
+        self.options = options
+
+    def dumps(self):
+        """Return path command representation."""
+
+        ret_str = self.path_type
+
+        if self.options is not None:
+            ret_str += self.options.dumps()
+
+        return ret_str
+
+
+class TikZPathList(LatexObject):
+    """Represents a path drawing."""
+
+    _legal_path_types = ['--', '-|', '|-', 'to',
+                         'rectangle', 'circle',
+                         'arc', 'edge']
+
+    def __init__(self, *args):
+        """
+        Args
+        ----
+        args: list
+            A list of path elements
+        """
+        self._last_item_type = None
+        self._arg_list = []
+
+        # parse list and verify legality
+        self._parse_arg_list(args)
+
+    def append(self, item):
+        """Add a new element to the current path."""
+        self._parse_next_item(item)
+
+    def _parse_next_item(self, item):
+
+        # assume first item is a point
+        if self._last_item_type is None:
+            try:
+                _item = self._parse_point(item)
```

Comment:
These three lines of code occur a lot here. Could you make it a method so it's not repeated all the time. Maybe change `_parse_point` and `_parse_path` to `_add_point` and `_add_path` (and make them non static).